eng/scripts/Publish-MavenPackages.ps1

```diff
@@ -145,6 +145,17 @@ foreach ($packageDetail in $packageDetails) {
   }
   elseif ($RepositoryUrl -like "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
     if (!$StageOnly) {
```

Comment:
This logic isn't right...

If StageOnly, we don't want to release a snapshot.  Staging snapshots doesn't appear to be supported.